src/aequitas/preprocessing.py

```diff
@@ -1,66 +1,62 @@
 import logging
-from sys import exit
-
 import pandas as pd
+from typing import List, Tuple
 
 logging.getLogger(__name__)
 
 __author__ = "Rayid Ghani, Pedro Saleiro <saleiro@uchicago.edu>"
 __copyright__ = "Copyright \xa9 2018. The University of Chicago. All Rights Reserved."
 
-def check_required_cols(df, required_cols):
-    """
 
+def check_required_cols(df: pd.DataFrame, required_cols: List[str]) -> None:
+    """
     :param df: A data frame of model results
     :param required_cols: Column names required for selected fairness measures
     :return: None, or ValueError
     """
-    check_model_cols = [col in df.columns for col in required_cols]
-    if False in check_model_cols:
-        raise ValueError
-    return
+    missing_cols = [col for col in required_cols if col not in df.columns]
+    if missing_cols:
+        raise ValueError(f"Missing required columns: {missing_cols}")
 
 
-def get_attr_cols(df, non_attr_cols):
+def get_attr_cols(df: pd.DataFrame, non_attr_cols: List[str]) -> List[str]:
     """
     :param df: A data frame of model results
     :param non_attr_cols: Names of columns not associated with attributes
     :return: List of columns associated with sample attributes
     """
-    # index of the columns that are associated with attributes
     attr_cols = df.columns[~df.columns.isin(non_attr_cols)]
     if attr_cols.empty:
-        raise ValueError
+        raise ValueError("No attribute columns found in the input data frame.")
     return attr_cols.tolist()
 
 
-def discretize(df, target_cols):
+def discretize(df: pd.DataFrame, target_cols: List[str]) -> pd.DataFrame:
     """
-
     :param df: A data frame of model results
     :param target_cols: Names of columns to discretize
     :return: A data frame
     """
     for col in target_cols:
-        if len(df[col].unique()) > 1:
+        unique_values = df[col].unique()
+        if len(unique_values) > 1:
             bins, values = pd.qcut(df[col], 4, precision=2, labels=False, duplicates='drop', retbins=True)
             try:
                 df[col] = bins.map(lambda x: '%0.2f' % values[x] + '-' + '%0.2f' % values[x + 1])
             except Exception as e:
-                logging.info('Something strange with a column in the input_df ' + str(e))
+                logging.info('Something strange with a column in the input_df: %s', e)
```

Comment:
It wouldn't hurt to apply this form to the other uses of f-strings with `logger` as well.